packages/server-side-render/src/server-side-render.js

```diff
@@ -88,19 +87,17 @@ export default function ServerSideRender( props ) {
 
 		setIsLoading( true );
 
-		const sanitizedAttributes =
-			attributes &&
-			__experimentalSanitizeBlockAttributes( block, attributes );
+		// The __internalWidgetId is only defined on the client-side and must be removed before making the
+		// request.
+		const retainedAttributes = omit( attributes, [ '__internalWidgetId' ] );
 
```

Comment:
We might be able to do this in a more generic way, but the `__internalWidgetId` is actually a special case and is the only attribute that needs to be stripped here 🤔 

My plan was to just add back in `__experimentalSanitizeBlockAttributes`, which on trunk handles the removal of the widgetId -- but that will no longer work! That function removes _un-registered_ attributes, but the widgetId _is_ registered now (just only on the client side).

We _don't_ want to omit all attributes with `copy: false` here, because it's entirely possible to add 'normal' server-side attributes that don't support copy, and those shouldn't be stripped. Only the widgetId is a special case.

Alternative suggestions from @noisysocks on #34750:
> We could just set 'additionalProperties' to true on the server. I don't really see the need to be so strict there. This might require a Core patch, though, as I'm not sure that it's possible for a plugin to re-register a route.
>
> Or, we could have ServerSideRender perform a GET on the /block-types endpoint before making its request to /block-renderer. If an attribute isn't in the attributes object that comes back from /block-types, then it hasn't been defined on the server, and so we must not send it to /block-renderer.

Reply:
> We could just set 'additionalProperties' to true on the server. I don't really see the need to be so strict there. This might require a Core patch, though, as I'm not sure that it's possible for a plugin to re-register a route.

I'm leaning towards this since it matches how the frontend works, i.e. you can put whatever you like in the block's markup and the block editor will silently ignore it.

But yeah this is fine is a temporary solution. Before merging, could you please create an issue to track this and link to it in a `// TODO`?